tiledb/sm/serialization/query.cc

```diff
@@ -1495,8 +1495,8 @@ Status query_from_capnp(
 
```

Comment:
uint64_t* existing_buffer_size_ptr = nullptr;
    uint64_t* existing_offset_buffer_size_ptr = nullptr;
    uint64_t* existing_validity_buffer_size_ptr = nullptr;

These are only used in the type == QueryType::WRITE cases and could be moved there.